runtime/str.go

```diff
@@ -753,6 +753,12 @@ func strInterpolate(f *Frame, format string, values *Tuple) (*Object, *BaseExcep
 					return nil, raised
 				}
 				val = s.Value()
+			} else if matches[7] == "o" {
+				if o.isInstance(LongType) {
+					val = toLongUnsafe(o).Value().Text(8)
```

Comment:
I realized the other day when I was looking at this code that CPython uses the __hex__ and __oct__ slots for this: https://github.com/python/cpython/blob/2.7/Objects/stringobject.c#L4019

So I think the right thing to do here is to implement those slots for int and long types. I think it would make this code considerably simpler.